dvc/command/data_sync.py

```diff
@@ -190,7 +190,7 @@ def add_parser(subparsers, _parent_parser):
         "--recursive",
         action="store_true",
         default=False,
-        help="Push cache from subdirectories of specified directory.",
+        help="Push cache for subdirectories of specified directory.",
```

Comment:
I guess it is still not clear that it will be pushed for dvc files inside of that directory, but it is still better.